src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -756,11 +783,12 @@ public struct RequestEventDispatcher
     }
     body
     {
-        foreach ( i, waiting_fiber; this.waiting_fibers )
-            if ( waiting_fiber.event.active == waiting_fiber.event.active.send )
+        foreach ( ref waiting_fiber; this.waiting_fibers )
+            if ( waiting_fiber.enabled &&
+                waiting_fiber.event.active == waiting_fiber.event.active.send )
             {
                 WaitingFiber ret = waiting_fiber;
```

Comment:
```
this.disable(waiting_fiber);
return waiting_fiber;
```

or am I missing something?

Reply:
I think the idea is to make a copy of it before it's disabled, then return the copy.

Reply:
(I don't remember what difference this could make, though.)

Reply:
Ah, could it be due the fact that the old code was moving this instance around the array, invalidating this reference to it? In that case, it's not needed any more.

Reply:
Maybe. I'll have a look at the usage.

Reply:
Yep, you're right. Looks like a refactoring omission. Updated.